### DISCUSSION TITLE
Mutate Value Of Subscribe Object Without Convert It To Observable

### DISCUSSION THREAD

**Discussion by mahdi-farnia** [ReactiveX/rxjs#6987] [Q&A]
hi, i'm new to RxJS and reactive programming. I asked my question in stackoverflow but i didn't get any response yet!

i want to use these two solutions together with RxJS but I don't know how to do it.
```ts
// Sample code, this does not work properly :(, because next is not defined on Observable
const dummy = new Subject<number>().pipe(
  map((num) => num + 1)
);

dummy.subscribe((number) => {
  // expects 4 but get 3
})

dummy.next(3)
```

short explanation is:
> I want to emit data everywhere even outside of observable construction like Subscribe, and operate on each data emit using pipe method like Observable.

I can implement a simple emitter class that simulate this behavior but I want a **RxJS way**.

BTW this is stackoverflow [question link](https://stackoverflow.com/questions/72502374/mutate-value-of-subscribe-object-without-convert-it-to-observable)

Thanks 🙏 

> Actually this question has been answered you can see the answer on stackoverflow website

**Comment by jakovljevic-mladen** [comment]
Would this work?

```ts
const dummy = new Subject<number>();

const result = dummy.pipe(map((num) => num + 1));

result.subscribe((number) => {
  // expects 4 but get 3
});

dummy.next(3)
```
